Makefile

```diff
@@ -23,14 +23,14 @@ PROJECT = $(shell git rev-parse --show-toplevel)
 INCLUDE = -I$(PROJECT) -I$(PROJECT)/mbedtls/include
 
 # Set our standard C++ compiler flags
-CXXFLAGS = -g -std=c++17 -fpic -DSQLITE_ENABLE_NORMALIZE $(BEDROCK_OPTIM_COMPILE_FLAG) -Wall -Werror -Wformat-security $(INCLUDE)
+CXXFLAGS = -g -std=c++17 -fpic -DSQLITE_ENABLE_NORMALIZE $(BEDROCK_OPTIM_COMPILE_FLAG) -Wall -Werror -Wformat-security  -Wno-error=deprecated-declarations $(INCLUDE)
 
 # All our intermediate, dependency, object, etc files get hidden in here.
 INTERMEDIATEDIR = .build
 
 # We use the same library paths and required libraries for all binaries.
 LIBPATHS =-L$(PROJECT) -Lmbedtls/library
-LIBRARIES =-lbedrock -lstuff -lbedrock -ldl -lpcrecpp -lpthread -lmbedtls -lmbedx509 -lmbedcrypto -lz -lm
+LIBRARIES =-Wl,--start-group -lbedrock -lstuff -Wl,--end-group -ldl -lpcrecpp -lpthread -lmbedtls -lmbedx509 -lmbedcrypto -lz -lm
```

Comment:
From the docs, I see this regarding `--start-group`/`--end-group`

> Using this option has a significant performance cost. It is best to use it only when there are unavoidable circular references between two or more archives.

Are their circular dependencies between libstuff and bedrock that have to be resolved using this?

Reply:
So you can see from the previous line we already had `-lbedrock -lstuff -lbedrock` and what I needed now was `-lbedrock -lstuff -lbedrock -lstuff` and I thought that was far enough toward absurdity to not do it anymore. To be clear, this is a compile-time performance cost for the linker, not a runtime performance cost.